src/controller/CHIPDeviceController.cpp

```diff
@@ -894,8 +895,13 @@ CHIP_ERROR DeviceCommissioner::StopPairing(NodeId remoteDeviceId)
 
 CHIP_ERROR DeviceCommissioner::UnpairDevice(NodeId remoteDeviceId)
 {
-    // TODO: Send unpairing message to the remote device.
-    return CHIP_NO_ERROR;
+    MATTER_TRACE_EVENT_SCOPE("UnpairDevice", "DeviceCommissioner");
+
+    if (mState != State::Initialized) // the device is currently being paired
```

Comment:
This part still does not make sense to me: if mState is not Initialized, we can't be worked with at all.  There _can't_ be a device being paired in this state.  If we're not initialized, we should just return error.

Reply:
I was mistaken about the meaning of the mState variable. I modified it.